src/components/button-toggle/button-toggle.component.tsx

```diff
@@ -74,47 +117,73 @@ export const ButtonToggle = ({
     );
   }
 
-  function handleClick() {
-    inputRef.current?.focus();
+  function handleClick(ev: React.MouseEvent<HTMLButtonElement>) {
+    if (onClick) {
+      onClick(ev);
+    }
+    if (onChange) {
+      const customEvent = { target: { name: groupName || name, value } };
+      onChange(customEvent);
+    }
+    if (value) {
+      onButtonClick(value);
+    }
+  }
+
+  function handleFocus(ev: React.FocusEvent<HTMLButtonElement>) {
+    if (onFocus) {
+      onFocus(ev);
+    }
+    if (inputGroupOnFocus) {
+      inputGroupOnFocus();
+    }
   }
 
+  function handleBlur(ev: React.FocusEvent<HTMLButtonElement>) {
+    if (onGroupBlur) {
+      onGroupBlur(ev);
+    } else if (onBlur) {
+      onBlur(ev);
+    }
+    if (inputGroupOnBlur) {
+      inputGroupOnBlur();
+    }
+  }
+
+  const isPressed =
+    pressedButtonValue === null
+      ? pressedPropValue
+      : pressedButtonValue === value;
+
   return (
-    <StyledButtonToggle
+    <StyledButtonToggleWrapper
       data-component={dataComponent || "button-toggle"}
```

Comment:
As we're making a breaking change with this PR, it might be worth refactoring this so it extends `TagProps` and removes the customisability of `data-component`. There are a few other components that have the same implementation so I'm happy for you to leave this as is and we can do them all in one ticket if that's preferable